Vineyard/Vineyard/GroupsListViewModel.swift

```diff
@@ -205,4 +205,42 @@ class GroupsListViewModel {
             }
         }
     }
+    
+    func leaveGroup(_ group: Group) {
+        guard var user = user else {
+            print("User is not set.")
+            return
+        }
+        
+        Task {
+            do {
+                // Remove group from user's groupIDs
+                user.groupIDs.removeAll { $0 == group.id }
+                
+                // Create updated group with user removed
+                var updatedGroup = group
+                updatedGroup.peopleIDs.removeAll { $0 == user.id }
+                
+                // Remove all progress entries related to this group's resolutions
+                if var allProgress = user.allProgress {
+                    user.allProgress = allProgress.filter { progress in
+                        !group.resolutionIDs.contains(progress.resolutionID)
+                    }
+                    user.allProgressIDs = user.allProgress?.map { $0.id ?? "" } ?? []
+                }
+                
+                // Update Firebase
+                try await databaseManager.addPersonToDB(person: user)
+                try await databaseManager.addGroupToDB(group: updatedGroup)
```

Comment:
Out of curiosity, how does this replace the existing group and not add a new one? Is it because they have the same document id?